File: src/Render.java
import java.awt.*;
import java.awt.geom.Line2D;
import java.awt.geom.Point2D;
import java.awt.image.BufferStrategy;

class Render{
    private long RENDER = System.currentTimeMillis();
    private CustomWars GAME = null;

    private int TICK = 20;

    Render(CustomWars game){
        GAME = game;
    }

    void render(){
        if(System.currentTimeMillis() - RENDER >= 1000 / TICK){
            RENDER = System.currentTimeMillis();
            BufferStrategy bs = GAME.getBufferStrategy();
            if (bs == null){
                GAME.createBufferStrategy(2);
                GAME.requestFocus();
                return;
            }
            Graphics2D graphics = (Graphics2D) bs.getDrawGraphics();
            graphics.setColor(Color.BLACK);
            graphics.fillRect(0, 0, GAME.getWidth(), GAME.getHeight());
            GAME.ENTITY.stream().filter(e -> e.TARGET != null).filter(e -> Point2D.distance(e.X, e.Y, e.TARGET.X, e.TARGET.Y) <= e.RADIUS_ATTACK).filter(e -> e.TARGET.HP > 0).forEach(e -> {
                graphics.setColor(e.COLOR_DRAW);
                graphics.draw(new Line2D.Double(e.X, e.Y, e.TARGET.X, e.TARGET.Y));
            });
            for(Entity e : GAME.ENTITY){
                e.render(graphics);
            }
            int green = 0;
            int orange = 0;
            for(Entity e : GAME.ENTITY){
                if(e.FACTION == Color.GREEN){
                    green++;
                }else orange++;
            }
            graphics.setColor(Color.WHITE);
            graphics.drawString("Green = " + green, 700, 18);
            graphics.drawString("Orange = " + orange, 693, 29);
            graphics.dispose();
            bs.show();
        }
    }
}

File: src/Render.kt
import java.awt.Color
import java.awt.Graphics2D
import java.awt.geom.Line2D
import java.awt.geom.Point2D

internal class Render {
    private var render = System.currentTimeMillis()

    private val tick = 20

    fun render() {
        if (System.currentTimeMillis() - render >= 1000 / tick) {
            render = System.currentTimeMillis()
            val bs = CustomWars.GAME.bufferStrategy
            if (bs == null) {
                CustomWars.GAME.createBufferStrategy(2)
                CustomWars.GAME.requestFocus()
                return
            }
            val graphics = bs.drawGraphics as Graphics2D
            graphics.color = Color.BLACK
            graphics.fillRect(0, 0, CustomWars.WIDTH, CustomWars.HEIGHT)
            CustomWars.entity.stream().filter({ e -> e.target != null && e is EntityLaser }).filter({ e -> Point2D.distance(e.x, e.y, e.target!!.x, e.target!!.y) <= e.radiusAttack && e.target!!.life && e.life }).filter({ e -> e.target!!.hp > 0 }).forEach { e ->
                graphics.color = e.faction
                graphics.draw(Line2D.Double(e.x, e.y, e.target!!.x, e.target!!.y))
            }
            for (e in CustomWars.entity) {
                if (e.life) e.render(graphics)
            }
            for (e in CustomWars.bullet) {
                if (e.life) e.render(graphics)
            }
            var green = 0
            var orange = 0
            for (e in CustomWars.entity) {
                if (e.life)
                    if (e.faction === Color.GREEN) {
                        green++
                    } else
                        orange++
            }
            graphics.color = Color.GREEN
            graphics.drawString("Green = $green", 700, 18)
            graphics.color = Color.ORANGE
            graphics.drawString("Orange = $orange", 693, 33)
            graphics.color = Color.WHITE
            graphics.drawString("Entity = ${CustomWars.entity.size}", 705, 48)
            graphics.drawString("Bullet = ${CustomWars.bullet.size}", 703, 63)
            graphics.dispose()
            bs.show()
        }
    }
}
